File: playwright-python-tests/page-objects/home_page.py
from .base_page import BasePage

class HomePage(BasePage):
    """HomePage represents GitHub.com main dashboard and open elements."""
    def __init__(self, page):
        super().__init__(page)
        self.url = "https://github.com"
        self.search_box = "input[placeholder*='Search']"
        self.star_button = "button:has-text('Star')"
        self.unstar_button = "button:has-text('Unstar')"
        self.profile_button = "img.avatar"
        self.explore_link = "a:has-text('Explore')"

    def search_repo(self, query: str) -> None:
        self.fill_input(self.search_box, query)
        self.page.keyboard.press("Enter")
        self.page.wait_for_load_state("networkidle")

File: playwright-python-tests/page-objects/HomePage.ts
import { Page, Locator } from '@playwright/test';

export class HomePage {
  readonly page: Page;
  readonly searchBox: Locator;
  readonly starButton: Locator;
  readonly unstarButton: Locator;
  readonly profileButton: Locator;
  readonly exploreLink: Locator;

  constructor(page: Page) {
    this.page = page;
    this.searchBox = page.locator("input[placeholder*='Search']");
    this.starButton = page.locator("button:has-text('Star')");
    this.unstarButton = page.locator("button:has-text('Unstar')");
    this.profileButton = page.locator('img.avatar');
    this.exploreLink = page.locator("a:has-text('Explore')");
  }

  async navigate(): Promise<void> {
    await this.page.goto('https://github.com', { waitUntil: 'networkidle' });
  }

  async searchRepo(query: string): Promise<void> {
    await this.searchBox.fill(query);
    await this.page.keyboard.press('Enter');
    await this.page.waitForLoadState('networkidle');
  }
}
